tools/osbuild-image-test

```diff
@@ -204,132 +221,143 @@ class TestCase:
 
         return True
 
+    def _detect_exports(self) -> List[str]:
+        pipelines = self.manifest.pipelines
+        if not pipelines:
+            return []
+        # for now we just assume the last pipeline is the export
+        return [next(reversed(pipelines))]
 
-class ImageTestSuite:
-    pass
-
-
-def load_tests(path) -> Iterator[TestCase]:
-    files = os.scandir(path)
-
-    for entry in files:
-        if not entry.is_file():
-            continue
-
-        fullpath = os.path.join(path, entry.name)
-        case = TestCase.load(fullpath)
-        yield case
-
-
-def detect_exports(test: TestCase) -> List[str]:
-
-    pipelines = test.manifest.pipelines
-    if not pipelines:
-        return []
+    def _detect_checkpoints(self) -> List[str]:
+        checkpoints = []
+        manifest = self.manifest
 
-    # for now we just assume the last pipeline is the export
-    return [next(reversed(pipelines))]
+        build = manifest.get("build")
+        if build:
+            checkpoints += [build.name]
 
+        commit = manifest.get("ostree-commit")
+        if commit:
+            checkpoints += [commit.name]
```

Comment:
We could checkpoint everything, which would be great for situations where manifests change in late stages only.

Future improvement though.